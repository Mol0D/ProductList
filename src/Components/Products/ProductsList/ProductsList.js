import React from 'react';
import './ProductsList';
import {Spinner, Container, Row, Col} from 'react-bootstrap';
import ProductCard from '../ProductCard/ProductCard';
import SearchForm from '../SearchForm/SearchForm';

const ProductsList = props =>{
    return(
        <div>
          <SearchForm filter={props.filter}/>
{
            props.isLoading ? 
              <div className="spinner">
                <Spinner animation="border" variant="primary" />
              </div>
               : 
              <Container>
                <Row>
                {props.data && props.data.map((item, index)=>{
                    const {name, img, price, stars, link, brand} = item;
                      return(
                        <Col xs={6} key={index}>
                        <ProductCard
                          name={name}
                          img={img}
                          price={price}
                          stars={stars}
                          link={link}
                          brand={brand}
                          />
                        </Col>
                      )
                    })
                   }
                  </Row>
                </Container>
              }
        </div>
        
    )
}

export default ProductsList;