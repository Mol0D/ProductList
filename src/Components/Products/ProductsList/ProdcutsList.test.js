import React from 'react';
import {shallow, mount} from 'enzyme';
import ProductsList from './ProductsList';

describe("ProductsList", () => {

    const props = {
        data: [{}, {}, {}],
        isLoading: true,
        filter: "some filter"
    }

    it("without component", () => {
        const wrapper = shallow(<ProductsList/>);
        expect(wrapper).toMatchSnapshot();
    })

    it("with component", () => {
        const wrapper = shallow(<ProductsList {...props}/>);
        expect(wrapper).toMatchSnapshot();
    })

    it("correct Prop Data", () => {
        const wrapper = mount(<ProductsList data={props.data}/>);
        expect(wrapper.prop('data')).toEqual([{}, {}, {}])
    })

    it("correct Prop isLoading", () => {
        const wrapper = mount(<ProductsList isLoading={props.isLoading}/>);
        expect(wrapper.prop('isLoading')).toEqual(true)
    })

    it("correct Prop Filter", () => {
        const wrapper = mount(<ProductsList filter={props.filter}/>);
        expect(wrapper.prop("filter")).toEqual("some filter")
    })

})