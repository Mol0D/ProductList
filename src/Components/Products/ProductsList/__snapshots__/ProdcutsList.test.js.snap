// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList with component 1`] = `ShallowWrapper {}`;

exports[`ProductsList without component 1`] = `ShallowWrapper {}`;
